docs/Testing.md

```diff
@@ -1,13 +1,144 @@
 <h1 align="center">Fastify</h1>
 
 ## Testing
+
 Testing is one of the most important parts of developing an application. Fastify is very flexible when it comes to testing and is compatible with most testing frameworks (such as [Tap](https://www.npmjs.com/package/tap), which is used in the examples below).
 
-<a name="inject"></a>
-### Testing with http injection
+Let's `cd` into a fresh directory called 'testing-example' and type `npm init -y` in our terminal.
+
+run `npm install fastify && npm install tap pino-pretty --save-dev`
+
+### Separating concerns makes testing easy
+
+ First we're going to separate our application code from our server code:
+
+app.js
+
+```js 
+const Fastify = require('fastify')
+
+function build() {
+  const fastify = Fastify({
+    logger: {
+      level: 'info',
+      prettyPrint: true
+    }
+  })
+  fastify.get('/', async (request, reply) => {
+    return { hello: 'world' }
+  })
+
+  return fastify
+}
+
+module.exports = build
+```
+
+server.js
+
+```js
+const server = require('./app')()
```

Comment:
```suggestion
const server = require('./app')({
  logger: {
    level: 'info',
    prettyPrint: true
  }
}|)
```

Reply:
Right, passing the opts as an argument makes a lot of sense. All set @mcollina!